tests/rest_api_helper.py

```diff
@@ -1,13 +1,18 @@
 import requests
 
+class RestApiHTTPError(Exception):
+    def __init__(self, status_code, *args: object) -> None:
+        super().__init__(*args)
+        self.status_code = status_code
+
```

Comment:
What is the purpose of the above class?

Reply:
This class defines exception type to be used in test.  This way I can check fail scenario for error code. See for example the function  "test_single_character_invalid_id"